indexing-service/src/main/java/io/druid/indexing/common/task/ParallelIndexTaskRunner.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package io.druid.indexing.common.task;
+
+import io.druid.indexer.TaskState;
+import io.druid.indexing.common.TaskToolbox;
+
+/**
+ * ParallelIndexTaskRunner is the actual task runner of {@link ParallelIndexSupervisorTask}. There is currently a single
+ * implementation, i.e. {@link SinglePhaseParallelIndexTaskRunner} which supports only best-effort roll-up. We can add
+ * more implementations in the future.
+ */
+public interface ParallelIndexTaskRunner
+{
+  TaskState run(TaskToolbox toolbox) throws Exception;
```

Comment:
This is not terribly descriptive. Can you explain the expectations of the `run` method, especially for future implementations vs current implementation?

Reply:
So, `ParallelIndexTaskRunner` is the one actually processing the task. It can have multiple implementations based on the distributed indexing algorithm described in https://github.com/apache/incubator-druid/issues/5543.